diag_manager/diag_util.F90

```diff
@@ -2499,6 +2499,86 @@ SUBROUTINE prepend_attribute_file(out_file, att_name, prepend_value, err_msg)
     END IF
   END SUBROUTINE prepend_attribute_file
 
+  !> @brief Updates indices based on presence/absence of input indices is, js, ks, ie, je, and ke.
+  ! Computes halo sizes in the I and J dimensions.
+  ! This routine is intended to be used in diag manager.
+  !> @return .false. if there is no error else .true.
+  function recondition_indices(indices, field, is_in, js_in, ks_in, &
+    ie_in, je_in, ke_in, err_msg) result(ierr)
+    integer, intent(inout) :: indices(12) !< Stores indices in order:
```

Comment:
I don't love this.  I think a derived type would be better here.  This array requires that you know what each member is, and that requires looking at this line where it's defined.  If you had a derived type, you would be able to write something like `indicies%fie` instead of having to know that fie is `indicies(9)`.  So why not use a derived type instead of the array?  Do you need them to be in order in memory?